### ISSUE TITLE
[coor.tica] the meaning of lag when using stride

### ISSUE THREAD

**Issue by gph82** [markovmodel/PyEMMA#238]
The wonderful thing about the stride option is that one can "quick-parametrize" at high strides without really throwing away the input data, e.g.
```
stride = 20
tica = coor.tica(inp, lag, ndim, stride = stride )
Y = tica.get_output()
```
and Y still has the full data (say, for bootstrapping later on, or for better resolved FESs etc)

However, the original "lag" is meaningless here, since 
acf(Y)(tau = lag)= \lambda is not true. For it to be true, one should
```
coor.tica(inp, np.round(lag/stride), ndim, stride = stride)
```
Since I think that "lag" should be defined so that the above relation is true, I propose to change this.
The hard way, not so transparent way:
* convert lag=np.round(lag/stride) internally (not very transparent, although we just need to inform the user in advance)

The easy way
* just mention in the docu that:
lag: int, the lag time, in multiples of the input time step * the stride

**Comment by gph82** [comment]
Dis what I mean:
```
lag = 500
neig = 10

strides = np.array([1, 2, 5, 10, 50, 100] )
strided_lag = lag/strides

tica_objs = []
for stride, ilag in zip(strides, strided_lag):
    
    print "stride %u, strided lag %u, rounded from %4.2f"%(stride, np.round(ilag), ilag)
    tica_objs.append(coor.tica(inp, np.round(ilag), ndim, stride = stride))

# Some reshaping for plotting purposes
lambdas = np.zeros((neig, len(strides))).T
for ii, i_tica in enumerate(tica_objs):
    lambdas[ii] = i_tica.eigenvalues[:neig]

myfig, myax = plt.subplots(2,1, sharex = True, figsize = (10, 10))
myax[0].plot(np.array(strides), lambdas, '--o');
#myax[0].set_xlabel('stride / frames')
myax[0].set_ylabel('$\lambda_i$')
myax[0].legend(['$\lambda_{%u}$'%(ii+1) for ii in xrange(neig)], loc = 0)

myax[1].plot(np.array(strides), -lag/np.log(np.abs(lambdas)), '--o');
myax[1].set_xlabel('stride / frames')
myax[1].set_ylabel('$t_i$ / frames')
myax[1].legend(['$t_{%u}$'%(ii+1) for ii in xrange(neig)], loc = 0)
myfig.tight_layout()
![strides](https://cloud.githubusercontent.com/assets/7518004/7120049/92226efa-e20a-11e4-9cda-b13e351fa585.jpg)

```

**Comment by gph82** [comment]
Which, in terms of autocorrelation functions, yields this. The higher the stride, the poorer the estimates for the the faster processes, because one looses *a lot* of information throwing away 100 frames (for processes with ITS of ca. 500 frames). Still, estimating the transformation with a stride of 100 will be much faster and have very little effect (which is what we want) on the transformed data.

![acfs](https://cloud.githubusercontent.com/assets/7518004/7125761/e36f2608-e233-11e4-9ce7-5bef2c91939b.png)

**Comment by gph82** [comment]
I am observing this:

``` python
itica = pyemma.coordinates.tica(isource, lag=10, stride=100)
### edited ####
/home/mi/gph82/programs/PyEmma/pyemma/coordinates/transform/tica.py in _estimate(self, iterable, **kw)
    254         if not partial_fit and not any(iterable.trajectory_lengths(self.stride) > self.lag):
    255             raise ValueError("None single dataset [longest=%i] is longer than"
--> 256                              " lag time [%i]." % (max(iterable.trajectory_lengths(self.stride)), self.lag))
    257 
    258         it = iterable.iterator(lag=self.lag, return_trajindex=False)

ValueError: None single dataset [longest=10] is longer than lag time [10].
```

However, 

``` python
isource.trajectory_lengths()
### array([1000])
```
- I thought the behavior was that " ... the time units of lag will be identical to the time units of the original trajectory." (quoting @fabian-paul in https://github.com/markovmodel/PyEMMA/pull/380#issuecomment-115329523
- If actually lag = lag \* stride, then the docs should say something more informative, like this: https://github.com/markovmodel/PyEMMA/pull/380/commits/982561f89f7e5779e2a0cfd18197c3d8f3c7d2e1#diff-9c1785d744d846b19ecb0d064b4e0d91R833

Any comments would be very much appreciated. Thanks!

**Comment by clonker** [comment]
Your 

```
source.trajectory_lengths()
```

gives back the unstrided trajectory lengths, however you have a stride argument of 100, i.e., the trajectory length is effectively 10. Also, your lag is 10, so the lagged chunk would be empty.

Indeed, the time units of lag are identical to the time units of the original trajectory.

**Comment by gph82** [comment]
yes I understand all of that, my point is that lag=10 changes meaning with stride=[1,10,100], and that should be stated somewhere

**Comment by fabian-paul** [comment]
I think that the behaviour that you observe was not intended. Lag = 10 with stride = 100 applied to a trajectory of length 1000 should give a "lagged" effective trajectory of length 9 if I'm right.

**Comment by gph82** [comment]
yupp, I seem to remember that the point behind not merging the PR #380 is that it was not needed. Hence my surprise.

So you're saying that it's a question of the condition   

``` python
if not partial_fit and not any(iterable.trajectory_lengths(self.stride) > self.lag):
     raise ValueError("None single dataset [longest=%i] is longer than"
                           " lag time [%i]." % (max(iterable.trajectory_lengths(self.stride)), self.lag))
```

 , but in principle TICA would be doing the right thing, right?

**Comment by fabian-paul** [comment]
Yes, the code you are quoting is wrong but TICA should do the right thing.

**Comment by gph82** [comment]
Thanks, I was very surprised about this.

**Comment by fabian-paul** [comment]
I think the right condition should be something like:
any(iterable.trajectory_lengths(stride=self.stride, skip=self.lag) > 0)

**Comment by fabian-paul** [comment]
@gph82 Can you give that a try?

**Comment by gph82** [comment]
well I'm on #732 now, but I will soon

**Comment by gph82** [comment]
Yeah that is doing the trick. I have prepared a PR here if we want to merge directly #733

**Comment by franknoe** [comment]
Is only this check the problem, or is there a problem with striding itself? Striding should indeed be applied to the input whenever `stride=...` is an estimation parameter.

**Comment by gph82** [comment]
I think the check was the only problem: 
![snap1](https://cloud.githubusercontent.com/assets/7518004/13847367/32d827de-ec4d-11e5-8af5-257b035970a7.png)

**Comment by gph82** [comment]
This should have been closed with PR #733
